BlackNectar/BlackNectar/StoresTableViewController.swift

```diff
@@ -256,8 +261,83 @@ extension StoresTableViewController {
             }
             
         }
-    
+        
     }
+    
+}
 
+extension StoresTableViewController {
+    
+    func setsEdgePanGesture() {
+        
+        edgeGesture.addTarget(self, action: #selector(self.handleRightEdge(gesture:)))
+        edgeGesture.edges = .right
+        edgeGesture.delegate = self
+        
+        self.view.addGestureRecognizer(edgeGesture)
+        
+    }
+    
+    func handleRightEdge(gesture: UIScreenEdgePanGestureRecognizer) {
+        
+        switch gesture.state {
+            
+        case .began, .changed:
+            
+            if !panningWasTriggered {
```

Comment:
Recommendation: Put this code in a different function. Right now there is a lot of code happening in a switch statement.